index.js

```diff
@@ -1,69 +1,83 @@
-const todoInput = document.getElementById('todoInput');
-const todoList = document.getElementById('todoList');
+(function () {
+  const todoInput = document.getElementById('todoInput');
+  const todoList = document.getElementById('todoList');
 
-const doneElement = (el) => { //
-  el.classList.toggle("done");
-  const changeEl = el.parentNode.id;
-  toDoList.toggleTask(changeEl);
-}
+  const listItemForCopy = document.querySelector('.for-copy').cloneNode(true);
 
-const clickDelete = (el) => {
-  const deteledId = el.parentNode.id;
-  toDoList.deleteTask(deteledId);
-  renderElements();
-}
+  document.querySelector('.for-copy').remove();
+  listItemForCopy.classList.remove('for-copy');
+
+  const getInitialDataStorage = () => {
+    const items = { ...localStorage };
+    const keys = Object.keys(items);
+    const prevTasks = [];
+    keys.forEach((key) => {
+      const task = JSON.parse(localStorage.getItem(key));
+      prevTasks.push(task);
+    });
+    toDoList.applyLocalData(prevTasks);
+  };
+
+  window.onload = getInitialDataStorage;
 
-const renderElements = () => {
-  todoList.innerHTML = '';//clear our previous list
-  const tasks = toDoList.getTasks();
-  tasks.forEach(item => { //create new list from our array
-    const deleteBtn = document.createElement('button');
-    deleteBtn.innerHTML = 'X';
-    deleteBtn.classList.add('deleteBtn');
-    deleteBtn.addEventListener("click", function (e) { //add event when user delete the task
-      clickDelete(this);
-    })
+  const doneElement = (task) => {
+    const item = JSON.parse(localStorage.getItem(task.id));
```

Comment:
super mixed logic. After a few features you won't be able to support this code :)